src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -40,14 +40,18 @@ import com.microsoft.hyperspace.index.{IndexLogEntry, LogicalPlanSignatureProvid
  */
 object FilterIndexRule extends Rule[LogicalPlan] with Logging {
   override def apply(plan: LogicalPlan): LogicalPlan = {
-    // FilterIndex rule looks for (Scan -> Filter -> Project) pattern to trigger
-    // a transformation. Currently, this rule replaces a relation with an index when:
-    //  1. The index covers all columns from the filter predicate and project, and
+    // FilterIndex rule looks for below patterns, in ordered manner, to trigger a transformation:
+    //  Pattern-1: Scan -> Filter -> Project
+    //  Pattern-2: Scan -> Filter
+    // Pattern-2 covers the case where project node is eliminated or not present. An example is
+    // when all columns are selected.
+    // Currently, this rule replaces a relation with an index when:
+    //  1. The index covers all columns from the filter predicate and output columns list, and
     //  2. Filter predicate's columns include the first 'indexed' column of the index.
     plan transform {
```

Comment:
let's be explicit and use `transformDown`

Reply:
Done.